### PR TITLE
Feat: (VariantCustomization) Allow user to customize variant weights

### PR DESCRIPTION
**Pull request by jaynish-zen**
Variant List:

image

Variant Edit:

image

Variant Modal with Error: 

image

### CODE REVIEW

**Comment by ivarconr** [comment]
Cool, thanks!. Will look in to this one of the coming days.

**Comment by ivarconr** [comment]
We also need to support weightType in the variant schema, I created a PR for that:
https://github.com/Unleash/unleash/pull/614

File: src/component/feature/variant/add-variant.jsx [hunk 1]
```diff
@@ -167,10 +186,33 @@ function AddVariant({ showDialog, closeDialog, save, validateName, editVariant,
...
+                        value={data.weight}
+                        error={error.weight}
+                        type="number"
+                        disabled={data.weightType !== 'fix'}
+                        onChange={setVariantValue}
+                    />
+                    <div>
+                        <Switch name="weightType" checked={isFixWeight} onChange={setVariantWeightType}>
+                            Fix
+                        </Switch>
```
Comment (ivarconr) [inline src/component/feature/variant/add-variant.jsx:213]: Maybe also clean up the view in general a bit, something along the lines of this:
image

File: src/component/common/util.js [hunk 2]
```diff
@@ -22,11 +24,35 @@ export const trim = value => {
 };
 
 export function updateWeight(variants, totalWeight) {
-    const size = variants.length;
-    const percentage = parseInt((1 / size) * totalWeight);
+    const { remainingPercentage, variableVariants } = variants.reduce(
```
[the remark this answers is not in the document]
Comment (jaynish-zen) [reply src/component/common/util.js:27]: True, It was the same before but I was getting code smell warnings from sonar so need to change variable naming inside reduce function.

File: src/component/feature/variant/add-variant.jsx [hunk 3]
```diff
@@ -167,10 +186,33 @@ function AddVariant({ showDialog, closeDialog, save, validateName, editVariant,
...
+                        placeholder=""
+                        value={data.weight}
+                        error={error.weight}
+                        type="number"
+                        disabled={data.weightType !== 'fix'}
+                        onChange={setVariantValue}
+                    />
+                    <div>
+                        <Switch name="weightType" checked={isFixWeight} onChange={setVariantWeightType}>
+                            Fix
```
[the remark this answers is not in the document]
Comment (jaynish-zen) [reply src/component/feature/variant/add-variant.jsx:212]: Yeah. make sense!

**Comment by jaynish-zen** [comment]
Thanks, @ivarconr for the feedback and backend PR! Backend PR looks good to me 👍. Updated as per review!

**Comment by jaynish-zen** [comment]
Here is the updated design.
image

**Pull request merged**